pkg/apis/core/validation/validation.go

```diff
@@ -3543,7 +3543,7 @@ func validateInitContainers(containers []core.Container, regularContainers []cor
 			}
 		}
 
-		if len(ctr.ResizePolicy) > 0 {
+		if !opts.AllowSidecarResizePolicy && len(ctr.ResizePolicy) > 0 {
 			allErrs = append(allErrs, field.Invalid(idxPath.Child("resizePolicy"), ctr.ResizePolicy, "must not be set for init containers"))
 		}
```

Comment:
This needs an `else if opts.AllowSidecarResizePolicy` case that validates the resize policy (calls `validateResizePolicy`)

Reply:
We are already calling `validateContainerCommon` at line no: 3501 which in turn is calling `validateResizePolicy`, so I don't think we need to call this function again?

Reply:
Ah, makes sense, thanks.